hazelcast-client/src/test/java/com/hazelcast/client/listeners/AbstractListenersOnReconnectTest.java

```diff
@@ -585,9 +581,6 @@ private ClientConfig getNonSmartClientConfig() {
     void onEvent(String event) {
         events.remove(event);
         eventCount.incrementAndGet();
-        if (events.isEmpty()) {
-            eventsLatch.countDown();
```

Comment:
When you removed this, you are only checking how many time the onEvent is triggered. What if it was triggered twice for the same event and none-for an expected event. The test will pass in that case as well. The check was ensuring that the removed events also should match the expected events one-to-one. After this change, are we missing this case?